### ISSUE TITLE
多进程实验上报 system_metric 采集 & 数量未对齐

### ISSUE THREAD

**Issue by Nexisato** [SwanHubX/SwanLab#1532]
### 确认项

- [x] 我已确认我遇到的问题在[文档](https://docs.swanlab.cn)中未有说明。
- [x] 我已在 [GitHub](https://github.com/SwanHubX/SwanLab/issues) 上搜索过类似或已有的问题。
- [x] 我已确认此问题并非由网络错误引起。

### 问题描述

多进程上报实验指标，目前存在如下几个问题：
1. 未显示指定 step 时，隐式 step 的共享 metric 计数非预期值（如 4进程各打 8 个点，预期 32 个点，实际上只有 24 个点
2. 多进程上报时，system 相关指标未被采集到
3. 不同进程如果有不同的 init config，在实验主页对应的 config info 应当以创建改实验的进程 config 为准; 针对系统信息页面采集的 pid 同理

### 必须的复现代码

核心代码是
```python
import swanlab
import random
import uuid
from multiprocessing import Process


def train_process(process_id: int, train_hash: str, epochs: int = 10):
    """
    单个训练进程

    Args:
        process_id: 进程ID (0-3)
        train_hash: 共享的实验ID
        epochs: 训练轮数
    """
    # 每个进程登录（必须要）
    swanlab.login(api_key="xxx")

    # 初始化SwanLab，所有进程使用相同的id
    swanlab.init(
        project="my-awesome-project",
        experiment_name=f"multi-process-run-{process_id}",
        id=train_hash,  # 关键：所有进程共享同一个ID
        parallel="shared",  # 关键：启用并行模式
        config={
            "learning_rate": 0.02,
            "architecture": "CNN",
            "dataset": "CIFAR-100",
            "epochs": epochs,
            "process_id": process_id
        }
    )

    # 每个进程使用不同的随机种子，产生不同的数据
    random.seed(process_id * 100)
    offset = random.random() / 5

    # 模拟训练
    for epoch in range(2, epochs):
        acc = 1 - 2 ** -epoch - random.random() / epoch - offset
        loss = 2 ** -epoch + random.random() / epoch + offset

        # 记录训练指标，添加进程标识以区分不同进程的数据
        swanlab.log({
            f"process_{process_id}/acc": acc,
            f"process_{process_id}/loss": loss,
            # 也可以记录一个共享指标，看看聚合效果
            "avg_acc": acc
        })

        print(f"Process {process_id}, Epoch {epoch}: acc={acc:.4f}, loss={loss:.4f}")

    swanlab.finish()
    print(f"Process {process_id} finished!")


def main():
    # 生成一个共享的实验ID
    train_hash = str(uuid.uuid4())
    print(f"Shared experiment ID: {train_hash}")

    # 创建4个进程
    num_processes = 4
    processes = []

    for i in range(num_processes):
        p = Process(target=train_process, args=(i, train_hash))
        processes.append(p)
        p.start()

    # 等待所有进程完成
    for p in processes:
        p.join()

    print("All processes finished!")


if __name__ == "__main__":
    main()
```

复现步骤：
1. 运行此代码即可
2. https://swanlab.cn/@nexisato/my-awesome-project/runs/0126adf0-077f-4a57-8d2d-ea513bf9e722/chart

### 错误信息

1. 

<img width="1440" height="1622" alt="Image" src="https://github.com/user-attachments/assets/b4da29ab-e141-4f32-b71d-1c27c3f16521" />
2. 

<img width="1264" height="478" alt="Image" src="https://github.com/user-attachments/assets/1c1037e4-0427-41f9-8dbc-6d5a28cf3f51" />

3. 

<img width="1592" height="1004" alt="Image" src="https://github.com/user-attachments/assets/b7953381-1373-4c94-a64b-fc053f866a86" />

4.

<img width="3242" height="1826" alt="Image" src="https://github.com/user-attachments/assets/eb75f7f4-7a87-4dac-b69c-2e607b559aab" />

**Comment by Nexisato** [comment]
server 相关改动： https://github.com/SwanHubX/SwanLab-Server/pull/937

**Comment by Nexisato** [comment]
针对问题 3，多进程竞态条件下如无 master 主进程负责收集 fork 出的子进程 pid 和 config info，在 SwanLab 目前只能以单进程运行的前提下，这几个指标以在数据库中第一个创建实验的进程为准

**Comment by Nexisato** [comment]
此外，针对问题 2，如果存在多台机器向同一个实验 id 中写入指标数据，即从多进程拆分为多物理节点，那么系统指标应当以谁为准(..
